core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -116,6 +122,15 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
     this.separatorArray = new Array(this.checkpointCount);
   }
 
+  restartExploration(): void {
+    this.editableExplorationBackendApiService.resetExplorationProgressAsync(
+      this.explorationId
+    ).then(() => {
+      // Required for the put operation to deliver data to backend.
+      this.windowRef.nativeWindow.location.reload();
```

Comment:
This could be done I think although it will be a bit tricky, but I am not sure if we want that kind of a behaviour. We want to give user a feeling of clearing progress and restarting the whole exploration altogether and I think reloading seems good from that perspective, but would love to hear your thoughts. Thanks!

Reply:
I think from the user perspective, it should be the same if you programatically restart the exp (instead of doing a full refresh of the page.) In fact, it will be a bit faster and will benefit users having low speed internet connectivity.

Reply:
Ok @kevintab95, in that case let's do it in a seperate PR, let's first get this functionality up, it will take some time to implement it and we need to get this PR merged in quickly to atleast get the checkpoints functionality up and then we can do the revisions in a seperate PR. I have created an issue for this https://github.com/oppia/oppia/issues/15419